week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -18,36 +18,113 @@ This means the order the songs are played in will be random, but each song will
 
  */
 
-
 class MusicPlayer {
-    // Add your code here
-
+  // Add your code here
+  constructor() {
+    this.playlist = [];
+    this.playing = false;
+    this.trackIndex = 0;
+  }
+  add(track, artist) {
+    let song = {};
+    song.name = track;
+    song.artist = artist;
+    this.playlist.push(song);
+  }
+
+  play(choice) {
+    if (this.playlist.length == 0) {
+      console.log("You have no songs in your playlist!");
+    } else {
+      if (choice) {
+        console.log(
+          `Currently playing:${this.playlist[choice].name} by ${this.playlist[choice].artist}`
+        );
+      } else {
+        console.log(
+          `Currently playing:${this.playlist[this.trackIndex].name} by ${
+            this.playlist[this.trackIndex].artist
+          }`
+        );
+      }
+    }
+  }
+  repeat() {
+    if (this.trackIndex === this.playlist.length - 1) {
+      this.trackIndex = 0;
+      this.play;
```

Comment:
Well, I guess you wanted to call the play function here. But without the parentheses, it won't be invoked.

Reply:
But you recognised well the logical correlation that after resetting the `trackIndex` value, play function should be called which is responsible to start the song.